devito/interfaces.py

```diff
@@ -18,8 +56,23 @@ def set_initializer(self, lambda_initializer):
         assert(callable(lambda_initializer))
         self.initializer = lambda_initializer
 
+    # function to generate a hased file name used for memmap file
+    def _generate_memmap_filenmae(self):
+        hash_string = self.name + str(randint(0, 100000000))
+        return self._hashing_function(hash_string).hexdigest()
+
+    # function to allocate memory for this data, if _disk_path is not None, a numpy memmap is used
+    # if not a numpy ndarray is used, by default _disk_path is None
     def _allocate_memory(self):
-        self.pointer = aligned(np.zeros(self.shape, self.dtype, order='C'), alignment=64)
+        self._disk_path = DenseData._disk_path
+        if self._disk_path == None:
+            self.pointer = aligned(np.zeros(self.shape, self.dtype, order='C'), alignment=64)
+        else:
+            f = self._disk_path + "/data_" + self.name + "_" + self._generate_memmap_filenmae()
+            self.pointer = aligned(np.memmap(filename=f, dtype=self.dtype, mode='w+', shape=tuple(self.shape), order='C'), alignment=64)
+            DenseData._memmap_file_list.append(f)
+            print("memmap file written to: " + f)
+            print("if the script call DenseData.remove_memmap_file() at the end, this file will be deleted automatically. if you have interupted this script, please go to " + self._disk_path + " to delete this file.")
```

Comment:
As a user I really don't want to see this message every time I run with a memmap.